# inflammatome_package_sandbox

Shiny app for the inflammatomeR package

Run the app by running the following code:

```
install.packages("shiny")
library(shiny)
runGitHub("inflammatome_package_sandbox", "opalasca")
```

R package accompanying the manuscript "The inflammatome: A meta-analysis of human genes regulated during inflammation"

inflammatomeR is a collection of functions that can be used to assess the presence of inflammation in transcriptomics or proteomics data. 
The analysis is based on a set of 2000 genes that appear to be generally upregulated in inflammatory conditions, the "inflammatome", as well as a core set of 100 genes, the "inflammation signature". 
The inflammatome genes were identified by performing a meta-analysis of publicly available case-control transcriptomics data from diseases involving inflammation. A ranked list of genes was generated by combining 
differential expression results between disease and control across multiple datasets. The inflammatome genes were selected as top 2000 genes in this list, and the inflammation signature, as top 100. 

There are two main use cases:
1. Assess to what extent inflammation genes are upregulated when comparing two conditions:
- Gene set enrichment analysis and visualisation: inflammatomeR helps in assessing the enrichment of the inflammation signature as well as other inflammation-related gene sets within a ranked list of results from differential expression analysis. 
- Volcano plot visualisation of the results list, highlighting the inflammatome genes; if many of the inflammatome genes are among the top hits, one can choose to interpret the results as being driven 
by inflammation; if the focus of the study is to understand more disease-specific processes, the inflammatome genes can be filtered out e.g. when selecting genes for further validation, biomarkers, drug targets, etc  
2. Compute sample-wise inflammation score: inflammatomeR can be used to calculate an inflammation score for each sample in a dataset, which represents the level of inflammation in that sample, relative to the other samples;  

Two types of input data:
- a table with results from differential expression/abundance analysis (e.g. limma, DESeq2, limma-voom, edgeR, etc) 
- a table with normalized gene/protein expression values (e.g. log2 normalized counts, rlog, vst, cpm, etc)

Gene/protein identifiers accepted: Ensembl, Entrez, gene symbol (HGNC), UniProt, RefSeq. The ids must be unique. 
Warning: if the data is transcript-level (e.g. RefSeq), with many transcript isoforms per gene, the data should be summarized to gene-level before using inflammatomeR.
